### ISSUE TITLE
Create the 'DataStore' interface class.

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/ddpdemo#7]
This class will define the interface method that we will use to write a block of data to a storage medium.

**Comment by bieryAtFnal** [comment]
This task will involve
- deciding on the signature of the write() method
- creating the C++ class for this interface (the [Queue interface in the appfwk repo](https://github.com/DUNE-DAQ/appfwk/blob/develop/include/appfwk/Queue.hpp) may be a useful example)
- if possible, creating some unit tests that verify the correct operation of the class

**Comment by bieryAtFnal** [comment]
I haven't yet marked this Issue as 'Ready for work', but if someone has time to play with this before our next meeting, that would be fine.  (Of course, it probably needs the results of Issues #4 and #5 as input.)

**Comment by bieryAtFnal** [comment]
I  added a preliminary version of this interface in include/ddpdemo/DataStore.hpp.  This is on the _feature/Issue7_DataStore_ branch (this branch was created from the _feature/Issue5_DataBlock_ branch).

I haven't verified that this first version compiles or runs yet (since it's just an interface, it's hard to test).  Although, maybe I'll make a FakeStorage concrete implementation that simply throws the data away.

**Comment by bieryAtFnal** [comment]
I merged the latest code from the _develop_ branch into the branch for this Issue (_feature/Issue7_DataStore_) and verified that it still compiled.
I then created a simple little DataStore implementation called TrashCanDataStore, which as the name implies, simply throws the data away (after printing out a message to say that).  This was a good thing since it helped me find some typos in the DataStore.hpp file.  The TrashCanDataStore isn't the prettiest code, and its write() method should probably move to a TrashCanDataStore.cpp file at some point.  Also, the sample instantiation of the TrashCanDataStore instance in SimpleDiskWriter should be changed at some point to be a dynamic creation of the DataStore instance based on what is provided in the configuration.
All of this code has been committed and pushed to the central repo.
